draft-ietf-quic-recovery.md

```diff
@@ -816,48 +816,91 @@ The minimum congestion window is the smallest value the congestion window can
+exponential growth of the congestion window.
+
+The sender MUST exit slow start and enter a recovery period when a packet is
+lost or when the ECN-CE count reported by its peer increases.
+
+A sender re-enters slow start any time the congestion window is less than the
+slow start threshold, which only occurs after persistent congestion is
+declared.
 
-## Recovery Period {#recovery-period}
+### Recovery {#recovery-period}
 
-A NewReno sender enters the recovery period when it detects loss or an ECN-CE
-mark is received. A recovery period ends when a packet sent during the recovery
-period is acknowledged.
+A NewReno sender enters a recovery period when it detects the loss of a packet
+or the ECN-CE count reported by its peer increases. A sender that is already in
+a recovery period stays in it and does not re-enter it.
 
 On entering a recovery period, a sender MUST set the slow start threshold to
-half the value of the congestion window at the moment that loss is detected. The
-congestion window MUST be set to the reduced value of the slow start threshold
-before exiting the recovery period.
+half the value of the congestion window when loss is detected. The congestion
+window MUST be set to the reduced value of the slow start threshold before
+exiting the recovery period.
 
-Implementations MAY set the congestion window immediately on entering a recovery
-period or use other mechanisms, such as Proportional Rate Reduction
-({{?PRR=RFC6937}}), to reduce it more gradually. If the congestion window is
-reduced immediately, a single packet can be sent prior to reduction. This speeds
-up loss recovery if the data in the lost packet is retransmitted and is similar
-to TCP as described in Section 5 of {{?RFC6675}}.
+Implementations MAY set the congestion window immediately upon entering a
+recovery period or use other mechanisms, such as Proportional Rate Reduction
+({{?PRR=RFC6937}}), to reduce the congestion window more gradually. If the
+congestion window is reduced immediately, a single packet can be sent prior to
+reduction. This speeds up loss recovery if the data in the lost packet is
+retransmitted and is similar to TCP as described in Section 5 of {{?RFC6675}}.
 
 The recovery period aims to limit congestion window reduction to once per round
-trip. Therefore during recovery, the congestion window remains unchanged
-irrespective of new losses or increases in the ECN-CE counter.
+trip. Therefore during a recovery period, the congestion window does not change
+in response to new losses or increases in the ECN-CE count.
+
+A recovery period ends and the sender enters congestion avoidance when a packet
+sent during the recovery period is acknowledged. This is slightly different
+from TCP's definition of recovery, which ends when the lost segment that
+started recovery is acknowledged ({{?RFC5681}}).
+
+### Congestion Avoidance
+
+A NewReno sender is in congestion avoidance any time the congestion window is
```

Comment:
```suggestion
A sender is in congestion avoidance any time the congestion window is
```

Reply:
This is specific to NewReno, so I would leave this as it is.

Reply:
How is this specific to NewReno? It's the definition of the ssthresh that you are in slow start when your cwnd is below that value.

Reply:
It's specific to a sender that uses slow start with an ssthresh variable. BBR does not have an ssthresh, for example. Since we're only defining NewReno in this document, there isn't anything incorrect with saying "NewReno sender", though it might be slightly more specific than you might like.